spark-jobs/src/main/scala/za/co/absa/enceladus/common/version/VersionExt.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.common.version
+
+import za.co.absa.commons.version.{NumericComponent, Version}
+
+object VersionExt {
+
+  implicit class RichVersion(version: Version) {
+    /**
+     * This strips the version to what the final version should look like - only keeping the leading numeric components, e.g.
+     * `1.2.3-rc.2` -> `1.2.3`
+     * @return
+     */
+    def finalVersion: Version = {
+      val leadingNumericComponents = version.components.takeWhile(_.isInstanceOf[NumericComponent])
+      Version(leadingNumericComponents: _*)
+    }
```

Comment:
This looks like something the commons could benefit having. What do you think, @wajda?

Reply:
Initially, I thought so, too, but this is only specific for the semver implementation that I am specifically using in the code. If we added this to `Version` that was initialized a simple, this would not work correctly, as 

`1.2.3-dev.finalVersion` would yield `1.2`, because `3-dev` is not a `NumericComponent`.
```scala
import za.co.absa.commons.version.Version
import za.co.absa.enceladus.common.version.VersionExt._

Version.asSimple("1.2.3-dev").finalVersion // Version(WrappedArray(NumericComponent(1), NumericComponent(2)))
```

So, I think the idea itself is valid, but making this actually correct would require more effort than it might seem at first glance.

Reply:
Agree with Deniel. Although nothing stops us from adding a new `SemVer` object that would hold this method and maybe a few others, that are specific to the semantic version spec.

Reply:
Created a proposal in Commons - AbsaOSS/commons#13
Let's discuss it there

Reply:
Meantime, to avoid exposing internal version component types I'd recommend to simply take the first 3 items, regardless of their types. It would be equally correct from the SemVer spec perspective. And it's also shorter.
```scala
Version(version.components.take(3): _*)
```

Reply:
Agreed, I will update it as you suggest. The proposal makes sense, too.

Reply:
Adjusted as outlined. Hope you find it reasonable, @wajda.